src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -51,11 +49,48 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
             override fun visitValueParameter(o: RsValueParameter) = checkValueParameter(holder, o)
             override fun visitVis(o: RsVis) = checkVis(holder, o)
             override fun visitBinaryExpr(o: RsBinaryExpr) = checkBinary(holder, o)
+            override fun visitCallExpr(o: RsCallExpr) = checkCallExpr(holder, o)
+            override fun visitMethodCallExpr(o: RsMethodCallExpr) = checkMethodCallExpr(holder, o)
         }
 
         element.accept(visitor)
     }
 
+    private fun checkMethodCallExpr(holder: AnnotationHolder, o: RsMethodCallExpr) {
+        val fn = o.reference.resolve() as? RsFunction ?: return
+        if (fn.unsafe != null) {
+            checkUnsafeCall(holder, o)
+        }
+    }
+
+    private fun checkCallExpr(holder: AnnotationHolder, o: RsCallExpr) {
+        val path = (o.expr as? RsPathExpr)?.path ?: return
+        val fn = path.reference.resolve() as? RsFunction ?: return
+        if (fn.unsafe != null) {
+            checkUnsafeCall(holder, o)
+        }
+    }
+
+    private fun checkUnsafeCall(holder: AnnotationHolder, o: RsExpr) {
+        val parent = o.ancestors
+            .drop(1)    // skip the expression itself
+            .filter {
+                when (it) {
+                    is RsBlockExpr -> it.unsafe != null
+                    is RsFunction -> true
+                    else -> false
+                }
+            }
+            .firstOrNull() ?: return
+
+        if (parent is RsBlock || (parent is RsFunction && parent.unsafe == null)) {
+            val annotation = holder.createErrorAnnotation(o, "Call to unsafe function requires unsafe function or block [E0133]")
+            annotation.registerFix(AddWrapUnsafeFix(o))
```

Comment:
There're cases when we can't apply such fix because it would break code:

```rust
unsafe fn pi() -> f64 { 3.14 }
fn main() {
    let s = pi() * 10;
}
```

I think we could only add this fix when the call expression's parent is `RsBlock`.